feat(helm): Add service to chart and prometheus examples

mjsmith1028 opened the pull request:
The goal of this PR is to add the ability to enable a service for metacontroller via helm and provide examples of how to consume prometheus metrics.

mjsmith1028 commented:
I added two flags `prometheus.serviceEnabled` and `prometheus.serviceMonitorEnabled` to allow the user to enable them independently to avoid coupling the implementation to prometheus operator. Ready for review again 👍

grzesuav commented:
hi @mjsmith1028 , thanks for PR. I think that `Service` should be delivered on flag, without referring specifically to `prometheus` (`Service` is regular k8s object).

Also, I would not see `ServiceMonitor` (external CRD) into this helm chart. Usually ops define some generic ServiceMonitor which pick up Services by labels

mjsmith1028 commented:
> hi @mjsmith1028 , thanks for PR. I think that `Service` should be delivered on flag, without referring specifically to `prometheus` (`Service` is regular k8s object).
> 
> Also, I would not see `ServiceMonitor` (external CRD) into this helm chart. Usually ops define some generic ServiceMonitor which pick up Services by labels

HI @grzesuav gotcha. I can update this PR to only include `Service` into metacontroller's helm chart and decouple the prometheus/servicemonitor responsibility.

mjsmith1028 commented:
Hi @grzesuav , I have refactored the PR. Now, it is just to enable a `Service` for metacontroller. I updated the examples to show how one would use the service for prometheus metrics.

The failing `Test / Release /Tests` is due to flakiness. I was able to run it successfully locally. I have updated https://github.com/metacontroller/metacontroller/issues/509 with additional details regarding the flakiness.

Please let me know if this PR looks okay to you.

mjsmith1028 commented:
Also, added fix for integration test flakiness: https://github.com/metacontroller/metacontroller/pull/531/files#diff-76ed074a9305c04054cdebb9e9aad2d818052b07091de1f20cad0bbac34ffb52R36 

This will run the tests sequentially to avoid to conflict between compositecontroller and decoratorcontroller tests.

grzesuav commented:
:tada: This PR is included in version 4.2.0 :tada:

The release is available on [GitHub release](https://github.com/metacontroller/metacontroller/releases/tag/v4.2.0)

Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot :package::rocket:
